### PR TITLE
test(queue): complete memory adapter coverage

### PR DESCRIPTION
**Pull request by flc1125**
## Summary
Complete unit coverage for the in-memory queue adapter without changing production behavior.

## Changes
- Cover canceled consumer creation and nil delivery access
- Verify delayed tasks wake on their timer and blocked consumers wake after enqueue
- Cover default dead-letter queues, metadata initialization, and stopped timers
- Increase module statement coverage from 94.8% to 100%

## Motivation
- Protect queue scheduling, notification, and dead-letter edge cases with deterministic, service-free tests

## Testing
- `go test -race -count=20 ./...`
- `go test -coverprofile=coverage.out ./...` (100.0% statements)
- `make lint/queue/adapter/memory`

### CODE REVIEW

File: queue/adapter/memory/queue_test.go [hunk 1]
```diff
@@ -176,6 +176,113 @@ func TestQueue_NilDeliveryOperationsAreNoop(t *testing.T) {
...
+	})
+	type result struct {
+		delivery queue.Delivery
+		err      error
+	}
+	receiving := make(chan struct{})
+	received := make(chan result, 1)
+	go func() {
+		close(receiving)
+		delivery, err := consumer.Receive(t.Context())
```
[the remark this answers is not in the document]
Comment (flc1125) [reply queue/adapter/memory/queue_test.go]: Addressed in 9be06232. The test now runs in a testing/synctest bubble and calls synctest.Wait before enqueueing, which guarantees Receive is durably blocked on the notification path. A second Wait lets the awakened goroutine settle before the result is asserted, with no wall-clock scheduling delays.
